vald/evm/gateway_tx_confirmation.go

```diff
@@ -20,21 +20,24 @@ func (mgr Mgr) ProcessGatewayTxConfirmation(event *types.ConfirmGatewayTxStarted
 		return nil
 	}
 
+	var vote *voteTypes.VoteRequest
+
 	txReceipt, err := mgr.GetTxReceiptIfFinalized(event.Chain, common.Hash(event.TxID), event.ConfirmationHeight)
 	if err != nil {
 		return err
 	}
-	if txReceipt == nil {
-		mgr.logger().Infof("broadcasting empty vote for poll %s", event.PollID.String())
-		_, err := mgr.broadcaster.Broadcast(context.TODO(), voteTypes.NewVoteRequest(mgr.proxy, event.PollID, types.NewVoteEvents(event.Chain)))
+	if txReceipt.Err() != nil {
+		vote = voteTypes.NewVoteRequest(mgr.proxy, event.PollID, types.NewVoteEvents(event.Chain))
 
-		return err
-	}
+		mgr.logger().Infof("broadcasting empty vote for poll %s: %s", event.PollID.String(), txReceipt.Err().Error())
+	} else {
+		events := mgr.processGatewayTxLogs(event.Chain, event.GatewayAddress, txReceipt.Ok().Logs)
+		vote = voteTypes.NewVoteRequest(mgr.proxy, event.PollID, types.NewVoteEvents(event.Chain, events...))
 
-	events := mgr.processGatewayTxLogs(event.Chain, event.GatewayAddress, txReceipt.Logs)
+		mgr.logger().Infof("broadcasting vote %v for poll %s", events, event.PollID.String())
+	}
```

Comment:
Would be nice to refactor this common logic somehow (`if txReceipt.Err() != nil { // vote empty } else { ... }` ), since it shows up in several different places. But this PR is big enough as is I think.

Reply:
That's what I figured as well. It's a good spot to stop ATM.